.circleci/config.yml

```diff
@@ -1,42 +1,217 @@
 # Ruby CircleCI 2.1 configuration file
 version: 2.1
 
-jobs:
-  build:
-    docker:
-      - image: cimg/ruby:3.0.2
-      - image: redis:6.2-alpine
+persist-to-workspace: &persist-to-workspace
+  persist_to_workspace:
+    root: /home/circleci
+    paths:
+      - .asdf
+      - .bashrc
+      - ./project
 
-    working_directory: ~/app
+attach-to-workspace: &attach-to-workspace
+  attach_workspace:
+    at: /home/circleci
 
-    steps:
-      - checkout
+executors:
+  ubuntu_vm:
+    machine:
+      image: ubuntu-2204:2024.04.4
```

Comment:
Do we need to hardcode the version?

Reply:
No, we could use `current`, but I think is better to stick to one version for stability.

https://circleci.com/developer/machine/image/ubuntu-2004